src/nih_wayfinding/app/scripts/geocoder/search-service.js

```diff
@@ -60,10 +59,26 @@
                     'f': 'pjson'
                 }
             }).success(function (data) {
-                dfd.resolve(searchToList(data));
-            }).error(function (data) {
-                dfd.resolve([]);
-                console.error('Geocoder.search(): ', data);
+                var response = searchToList(data);
+                if (response.length) {
+                    // check if location is within graph bounds
+                    MapControl.getGraphBounds().then(function(geojson) {
+                        var geom = response[0].geometry;
+                        // must use turf's geojson-y objects for turf.inside
+                        var point = turf.point([geom.x, geom.y]);
+                        var polygon = turf.polygon(geojson.coordinates);
+                        if (turf.inside(point, polygon)) {
+                            dfd.resolve(response);
+                        } else {
+                            dfd.reject('Address is outside the routing bounds. Please choose a different address.');
+                        }
+                    });
```

Comment:
Sorry I missed this earlier, but we should probably handle the case here where getGraphBounds fails, and just return the response without doing the graph bounds check.